Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM ubuntu:20.04 as builder
+ENV DEBIAN_FRONTEND noninteractive
+ENV DEBCONF_NONINTERACTIVE_SEEN true
+RUN echo 'tzdata tzdata/Areas select Etc' | debconf-set-selections; \
+    echo 'tzdata tzdata/Zones/Etc select UTC' | debconf-set-selections; \
+    apt-get update && apt-get upgrade -y && apt-get install --no-install-recommends -y \
+    cmake \
+    ninja-build \
+    gcc \
+    g++ \
+    maven \
+    flex \
+    bison \
+    libxml2-utils \
+    dpkg-dev
+COPY . /tmp/cbmc
+WORKDIR /tmp/cbmc
+RUN cmake -S . -Bbuild -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=/usr/bin/gcc -DCMAKE_CXX_COMPILER=/usr/bin/g++ && cd build; ninja -j2
+
+FROM ubuntu:20.04 as runner
+COPY --from=builder /tmp/cbmc/build/bin/* /tmp/cbmc/
```

Comment:
I still don't completely understand why this goes into `/tmp` rather than `/usr/bin`.

Reply:
Fair point - I can make that happen. I started with putting the binaries under `/home` but this was weird looking, and then it was decided it might be better to be put under `/tmp`.

I agree `/usr/bin` is a more appropriate location - I have this on my todo as part of finalising this PR.

Reply:
We should probably copy the built binaries to /usr/local/bin as this is usually the standard directory on Linux for binaries which are not managed by the package manager. It also has the advantage of being in the path. (I think we were using /tmp/cbmc for testing purposes yesterday.)